.circleci/config.yml

```diff
@@ -1,46 +1,40 @@
 version: 2
 jobs:
   build:
-    working_directory: /go/src/github.com/qlik-oss/enigma-go
     docker:
       - image: circleci/golang:1.11
     steps:
       - checkout
       - run:
-          name: Install toolchain
-          command: |
-            go get -u github.com/golang/dep/cmd/dep
-            go get -u golang.org/x/lint/golint
-      - restore_cache:
-          key: dependency-cache-{{ checksum "Gopkg.lock" }}
-      - run:
-          name: Install dependencies
-          command: dep ensure
-      - save_cache:
-          key: dependency-cache-{{ checksum "Gopkg.lock" }}
-          paths:
-            - vendor
+          name: Install lint
+          command: go get -u golang.org/x/lint/golint
       - run:
           name: Linting
           command: golint -set_exit_status
+      - restore_cache:
+          key: dependency-cache-{{ checksum "go.sum" }}
       - run:
           name: Build
           command: go build
+      - save_cache:
+          key: dependency-cache-{{ checksum "go.sum" }}
+          paths:
+            - "/go/pkg/mod"
       - run:
           name: Test
           command: go test -v -race ./...
       - setup_remote_docker
       - run:
           name: Execute integration tests
-          working_directory: /go
           command: |
             ENGINE_VERSION=$(curl -s "https://registry.hub.docker.com/v2/repositories/qlikcore/engine/tags/" | docker run -i stedolan/jq -r '."results"[0]["name"]' 2>/dev/null)
             echo "Will start Qlik Associative Engine version '$ENGINE_VERSION'"
             ENGINE_CONTAINER_ID=$(docker run -d qlikcore/engine:$ENGINE_VERSION -S AcceptEULA=yes)
             ENGINE_IP=$(docker inspect -f '{{ .NetworkSettings.IPAddress }}' $ENGINE_CONTAINER_ID)
-            docker cp /go/src/github.com/qlik-oss/enigma-go/examples/reload/monitor-progress/testdata/ $ENGINE_CONTAINER_ID:/testdata
-            TEST_CONTAINER_ID=$(docker run -d golang:1.11-alpine tail -f /dev/null)
-            docker cp . $TEST_CONTAINER_ID:/go
+            docker cp ./examples/reload/monitor-progress/testdata/ $ENGINE_CONTAINER_ID:/testdata
+            TEST_CONTAINER_ID=$(docker run -e "CGO_ENABLED=0" -d golang:1.11-alpine tail -f /dev/null)
```

Comment:
Had to disable `cgo`in test container since the modules support required gcc in the container. https://github.com/golang/go/wiki/Modules#why-does-go-build-require-gcc-and-why-are-prebuilt-packages-such-as-nethttp-not-used